src/EventBus/Message.php

```diff
@@ -71,112 +57,15 @@ public function event(): object
         return $this->event;
     }
 
-    /** @throws HeaderNotFound */
-    public function aggregateName(): string
-    {
-        $value = $this->headers[self::HEADER_AGGREGATE_NAME] ?? null;
-
-        if ($value === null) {
-            throw HeaderNotFound::aggregateName();
-        }
-
-        return $value;
-    }
-
-    public function withAggregateName(string $value): self
-    {
-        return $this->withHeader(self::HEADER_AGGREGATE_NAME, $value);
-    }
-
-    /** @throws HeaderNotFound */
-    public function aggregateId(): string
-    {
-        $value = $this->headers[self::HEADER_AGGREGATE_ID] ?? null;
-
-        if ($value === null) {
-            throw HeaderNotFound::aggregateId();
-        }
-
-        return $value;
-    }
-
-    public function withAggregateId(string $value): self
-    {
-        return $this->withHeader(self::HEADER_AGGREGATE_ID, $value);
-    }
-
     /**
-     * @return positive-int
+     * @param class-string<H> $name
+     *
+     * @return H
      *
      * @throws HeaderNotFound
+     *
+     * @template H of Header
      */
-    public function playhead(): int
-    {
-        $value = $this->headers[self::HEADER_PLAYHEAD] ?? null;
-
-        if ($value === null) {
-            throw HeaderNotFound::playhead();
-        }
-
-        return $value;
-    }
-
-    /** @param positive-int $value */
-    public function withPlayhead(int $value): self
-    {
-        return $this->withHeader(self::HEADER_PLAYHEAD, $value);
-    }
-
-    /** @throws HeaderNotFound */
-    public function recordedOn(): DateTimeImmutable
-    {
-        $value = $this->headers[self::HEADER_RECORDED_ON] ?? null;
-
-        if ($value === null) {
-            throw HeaderNotFound::recordedOn();
-        }
-
-        return $value;
-    }
-
-    public function withRecordedOn(DateTimeImmutable $value): self
-    {
-        return $this->withHeader(self::HEADER_RECORDED_ON, $value);
-    }
-
-    public function newStreamStart(): bool
-    {
-        $value = $this->headers[self::HEADER_NEW_STREAM_START] ?? null;
-
-        if ($value === null) {
-            throw HeaderNotFound::newStreamStart();
-        }
-
-        return $value;
-    }
-
-    public function withNewStreamStart(bool $value): self
-    {
-        return $this->withHeader(self::HEADER_NEW_STREAM_START, $value);
-    }
-
-    public function archived(): bool
-    {
-        $value = $this->headers[self::HEADER_ARCHIVED] ?? null;
-
-        if ($value === null) {
-            throw HeaderNotFound::archived();
-        }
-
-        return $value;
-    }
-
-    public function withArchived(bool $value): self
-    {
-        return $this->withHeader(self::HEADER_ARCHIVED, $value);
-    }
-
-    /** @throws HeaderNotFound */
     public function header(string $name): mixed
```

Comment:
```suggestion
    public function header(string $name): Header
```

Maybe we should generally adjust the API to return null if the header doesn't exist.
This is much more developer friendly.

```php
    public function header(string $name): Header|null
```

From this:

```php
        foreach ($messages as $message) {
            try {
                $newStreamStartHeader = $message->header(NewStreamStartHeader::class);
            } catch (HeaderNotFound) {
                continue;
            }

            if (!$newStreamStartHeader->newStreamStart) {
                continue;
            }

            $lastMessageWithNewStreamStart = $message;
        }
```

We can do this:

```php
        foreach ($messages as $message) {
            if (!$message->header(NewStreamStartHeader::class)?->newStreamStart) {
                continue;
            }
        }
```